pkg/oci/builder.go

```diff
@@ -63,52 +57,60 @@ func NewBuilder(name string, verbose bool) *Builder {
 //
 //	.func/builds/last
 func (b *Builder) Build(ctx context.Context, f fn.Function) (err error) {
-	cfg := &buildConfig{ctx, f, time.Now(), b.verbose, ""}
+	cfg := &buildConfig{ctx, b.name, f, time.Now(), b.verbose, "", b.tester, defaultPlatforms}
 
 	if err = setup(cfg); err != nil { // create directories and links
 		return
 	}
 	defer teardown(cfg)
 
-	//TODO: Use scaffold package when merged:
-	/*
-		if err = scaffolding.Scaffold(ctx, f, cfg.buildDir()); err != nil {
-			return
-		}
-	*/
-	// IN the meantime, use an airball mainfile
-	data := `
-package main
-
-import "fmt"
+	// Load the embedded repository
+	repo, err := fn.NewRepository("", "")
+	if err != nil {
+		return
+	}
 
-func main () {
-  fmt.Println("Hello, world!")
-}
-`
-	if err = os.WriteFile(path(cfg.buildDir(), "main.go"), []byte(data), 0664); err != nil {
+	// Write out the scaffolding
+	err = scaffolding.Write(cfg.buildDir(), f.Root, f.Runtime, f.Invoke, repo.FS())
+	if err != nil {
 		return
 	}
 
+	// Create an OCI container from the scaffolded function
 	if err = containerize(cfg); err != nil {
 		return
 	}
-	return updateLastLink(cfg)
+
+	if err = updateLastLink(cfg); err != nil {
+		return
+	}
 
 	// TODO: communicating build completeness throgh returning without error
 	// relies on the implicit availability of the OIC image in this process'
 	// build directory.  Would be better to have a formal build result object
 	// which includes a general struct which can be used by all builders to
 	// communicate to the pusher where the image can be found.
+	// Tests, however, can use a simple channel:
+	if cfg.tester != nil && cfg.tester.notifyDone {
+		if cfg.verbose {
+			fmt.Println("tester configured to notify on done.  Sending to unbuffered doneCh")
+		}
+		cfg.tester.doneCh <- true
+		fmt.Println("send to doneCh complete")
```

Comment:
This seems like debugging output or perhaps should be wrapped in `if cfg.verbose`

Reply:
Yes this only executes if there is a `tester`, so it will only print in the tests.